### PR TITLE
Add goto new functions

### PR DESCRIPTION
**Pull request by glannuzel**
Add functions to the server to be able to return more info the the user when using goto.

reachy-sdk-api must be on branch `add-goto-new-functions` as well. 
Descriptions of the expected results are in reachy-sdk-api PR.

### CODE REVIEW

File: reachy_sdk_server/reachy_sdk_server/grpc_server/goto.py [hunk 1]
```diff
@@ -280,6 +324,97 @@ def get_interpolation_mode(self, request: GoToRequest) -> str:
...
+                joints_goal=JointsGoal(neck_joint_goal=neck_joint_goal),
+                interpolation_mode=GoToInterpolation(interpolation_type=mode),
+            )
+
+            return request
+
+        else:
+            context.abort(grpc.StatusCode.NOT_FOUND, f"GoalId not found (id={goal_id}).")
+
     def cancel_goal_by_goal_id(self, goal_id: int) -> None:
```
Comment (FabienDanieau) [inline reachy_sdk_server/reachy_sdk_server/grpc_server/goto.py]: decoration is misleading -> bool

File: reachy_sdk_server/reachy_sdk_server/grpc_server/goto.py [hunk 2]
```diff
@@ -280,6 +324,97 @@ def get_interpolation_mode(self, request: GoToRequest) -> str:
             )
             return None
 
+    def _get_grpc_interpolation_mode(self, interpolation_mode: str) -> InterpolationMode:
```
Comment (FabienDanieau) [inline reachy_sdk_server/reachy_sdk_server/grpc_server/goto.py:327]: Optional[InterpolationMode] if None can be returned

**Review by FabienDanieau — APPROVED** [review]
Just some minor comments. Looks good

File: reachy_sdk_server/reachy_sdk_server/grpc_server/goto.py [hunk 3]
```diff
@@ -280,7 +324,98 @@ def get_interpolation_mode(self, request: GoToRequest) -> str:
...
+                f"Interpolation mode {interpolation_mode} not supported. Should be one of 'linear' or 'minimum_jerk'."
+            )
+            return None
+
+    def get_part_queue(self, part_name: str) -> GoToQueue:
+        goal_ids_int = getattr(self.goal_manager, part_name + "_goal")
+        goal_ids = [
+            GoToId(id=goal_id_int)
+            for goal_id_int in goal_ids_int
+            if self.goal_manager.goal_handles[goal_id_int].status in [0, 1]
```
Comment (RemiFabre) [inline reachy_sdk_server/reachy_sdk_server/grpc_server/goto.py:343]: The usage of magic numbers to compare a status is confusing. What does being 0 or 1 mean ?

File: reachy_sdk_server/reachy_sdk_server/grpc_server/goto.py [hunk 4]
```diff
@@ -280,7 +324,98 @@ def get_interpolation_mode(self, request: GoToRequest) -> str:
...
+        ]
+        return GoToQueue(goto_ids=goal_ids)
+
+    def get_part_goto_playing(self, part_name: str) -> GoToId:
+        goal_ids = getattr(self.goal_manager, part_name + "_goal")
+        for goal_id in goal_ids:
+            if self.goal_manager.goal_handles[goal_id].status == 2:
```
Comment (RemiFabre) [inline reachy_sdk_server/reachy_sdk_server/grpc_server/goto.py:350]: Same here for the == 2

File: reachy_sdk_server/reachy_sdk_server/grpc_server/goto.py [hunk 5]
```diff
@@ -260,11 +297,18 @@ def goto_joints(self, part_name, joint_names, goal_positions, duration, mode="mi
...
+        goal_request["duration"] = duration
+        goal_request["mode"] = mode
+
         if goal_handle is None:
             self.logger.info("GotoGoal was rejected")
             return GoToId(id=-1)
 
-        goal_id = self.goal_manager.store_goal_handle(goal_handle)
+        if part_name == "neck":
+            part_name = "head"
```
Comment (RemiFabre) [inline reachy_sdk_server/reachy_sdk_server/grpc_server/goto.py:310]: This scares me. Could you please add a comment in the code that explains the change in naming conventions?

**Comment by RemiFabre** [comment]
lol I'm a bit late to the party. My comments are minor, feel free to ignore

**Pull request merged**